uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt311/client/sdkmqtt/Mqtt311ConnectionImpl.java

```diff
@@ -84,14 +89,34 @@ public Mqtt311ConnectionImpl(@NonNull MqttLib.ConnectionParams connectionParams,
     @Override
     public ConnectResult start(long timeout, int connectionId) throws MqttException {
         this.connectionId = connectionId;
-        return null;
+
+        CompletableFuture<Boolean> connectFuture = connection.connect();
+        try {
+            Boolean sessionPresent = connectFuture.get(timeout, TimeUnit.SECONDS);
+            isConnected.set(true);
+            logger.atInfo().log("MQTT 3.1.1 connection {} is establisted", connectionId);
+            return buildConnectResult(true, sessionPresent);
+        } catch (TimeoutException | InterruptedException | ExecutionException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

Reply:
Fixed in 06dc2004c1859ce8fa9c0c6509c3c3ef3c891932